packages/ringcentral-integration/modules/ActiveCallControl/getDataReducer.js

```diff
@@ -0,0 +1,139 @@
+import { combineReducers } from 'redux';
+import { isHangUp, isReject } from './helpers';
+
+function updateActiveSessionStatus({
+  state,
+  party,
+  sessionId
+}) {
+  const newState = { ...state };
+  const {
+    muted,
+    standAlone,
+    status: {
+      code
+    },
+    direction,
+  } = party;
+  if (isHangUp(code) && newState[sessionId]) {
+    delete newState[sessionId];
+  } else {
+    newState[sessionId] = {
+      ...newState[sessionId],
+      muted,
+      standAlone,
+      code,
+      isReject: isReject({ direction, code })
+    };
+  }
+  return newState;
+}
+
+function getActiveSessionIdReducer(types) {
+  return (state = null, { type, sessionId }) => {
+    switch (type) {
+      case types.setActiveSessionId: {
+        return sessionId;
+      }
+      case types.resetSuccess: {
+        return null;
+      }
+      default:
+        return state;
+    }
+  };
+}
+function getRecordingIdsStatusReducer(types) {
+  return (state = {}, { type, sessionId, response }) => {
+    switch (type) {
+      case types.startRecord: {
+        const newState = { ...state };
+        newState[sessionId] = {
+          ...response
+        };
+        return newState;
+      }
+      case type.recordFail:
+      case types.resetSuccess:
+        return {};
+      default:
+        return state;
+    }
+  };
+}
+function getActiveSessionsStatusReducer(types) {
+  return (state = {}, {
+    type, sessionId, party, activeSessionsMap
+  }) => {
+    switch (type) {
+      case types.updateActiveSessions: {
+        let newState = null;
+        for (const sessionId in activeSessionsMap) {
+          if (sessionId) {
+            newState = updateActiveSessionStatus({
+              state,
+              party: activeSessionsMap[sessionId],
```

Comment:
the data structure of `party` does not have the `sessionId` filed?
if it has, then there is no need to pass the `sessionId` as the parameter,
if it has not, plz add the `filed`, since it should be a two way relationship